skpro/regression/linear/_glm.py

```diff
@@ -304,15 +539,82 @@ def _predict(self, X):
         -------
         y : pandas DataFrame, same length as `X`, with same columns as y in fit
         """
+        offset_var = self._offset_var
+        exposure_var = self._exposure_var
+        offset_arr = None
+        exposure_arr = None
+
+        if offset_var is not None:
+            if isinstance(offset_var, pd.Index):
+                offset_arr = np.array(X[offset_var]).flatten()
+            elif isinstance(offset_var, int):
+                offset_arr = np.array(X.iloc[:, offset_var]).flatten()
+                offset_var = pd.Index([X.iloc[:, offset_var].name])
+        if exposure_var is not None:
+            if isinstance(exposure_var, pd.Index):
+                offset_arr = np.array(X[exposure_var]).flatten()
+            elif isinstance(exposure_var, int):
+                exposure_arr = np.array(X.iloc[:, exposure_var]).flatten()
+                exposure_var = pd.Index([X.iloc[:, exposure_var].name])
+        if offset_var is not None and exposure_var is not None:
+            X = X.drop([offset_var[0], exposure_var[0]], axis=1)
+        elif offset_var is not None:
+            X = X.drop(offset_var, axis=1)
+        elif exposure_var is not None:
+            X = X.drop(exposure_var, axis=1)
+
         X_ = self._prep_x(X)
 
         index = X_.index
         y_column = self.y_col
-        y_pred_series = self.glm_fit_.predict(X_)
+        y_pred_series = self.glm_fit_.predict(
+            X_, offset=offset_arr, exposure=exposure_arr
+        )
         y_pred = pd.DataFrame(y_pred_series, index=index, columns=y_column)
 
         return y_pred
 
+    def _params_sm_to_skpro(self, y_predictions_df, index, columns, family):
+        """Convert the statsmodels output to equivalent skpro distribution."""
+        from skpro.distributions.gamma import Gamma
+        from skpro.distributions.normal import Normal
+        from skpro.distributions.poisson import Poisson
+
+        skpro_distr = {
+            "Normal": Normal,
+            "Poisson": Poisson,
+            "Gamma": Gamma,
+        }
+
+        params = {}
+        skp_dist = Normal
+
+        if family in skpro_distr:
+            skp_dist = skpro_distr[family]
+
+        if skp_dist == Normal:
+            y_mu = y_predictions_df["mean"].rename("mu").to_frame()
+            y_sigma = y_predictions_df["mean_se"].rename("sigma").to_frame()
+            params["mu"] = y_mu
+            params["sigma"] = y_sigma
+        elif skp_dist == Poisson:
+            y_mu = y_predictions_df["mean"].rename("mu").to_frame()
+            params["mu"] = y_mu
+        elif skp_dist == Gamma:
+            y_mean = y_predictions_df["mean"]
+            y_sd = y_predictions_df["mean_se"]
+            y_alpha = (y_mean / y_sd) ** 2
+            y_beta = (y_alpha / y_mean).rename("beta").to_frame()
```

Comment:
slightly numerically more stable would be `y_mean / (y_sd **2)` since you avoid error propagation due to cancellation, but this is very minor impact imo